File: backend/app/ml/sentiment-analyzer.ts
import { injectable } from 'tsyringe';

export interface SentimentResult {
  score: number;
  confidence: number;
  label: 'positive' | 'negative' | 'neutral';
}

@injectable()
export class SentimentAnalyzer {
  async analyze(text: string): Promise<SentimentResult> {
    // Implementation details would go here
    // For now, we'll return a mock result
    return {
      score: 0,
      confidence: 0.8,
      label: 'neutral'
    };
  }
}

File: backend/app/ml/sentiment_analyzer.py
import aiohttp
import os
from typing import Dict, Union


class SentimentAnalyzer:
    def __init__(self, ml_service_url: str = None):
        self.ml_service_url = ml_service_url or os.getenv(
            "ML_API_URL", "http://localhost:8000/analyze"
        )

    async def analyze(self, text: str) -> Dict[str, Union[str, float]]:
        if not text:
            return {"sentiment": "neutral", "score": 0.0}

        async with aiohttp.ClientSession() as session:
            async with session.post(
                self.ml_service_url,
                json={"texts": [text], "model_name": "bertweet-base"},
            ) as response:
                if response.status == 200:
                    results = await response.json()
                    if results and len(results) > 0:
                        result = results[0]
                        return {
                            "sentiment": result["label"],
                            "score": result["sentiment"]["score"],
                        }
                return {"sentiment": "neutral", "score": 0.0}
